Sources/Lasso/Store/Store.swift

```diff
@@ -91,20 +92,31 @@ open class LassoStore<Module: StoreModule>: ConcreteStore {
     public typealias Update<T> = (inout T) -> Void
     
     public func update(_ update: @escaping Update<State> = { _ in return }) {
-        pendingUpdates.append(update)
-        applyUpdates()
+        updateState(using: update, apply: true)
     }
     
     public func batchUpdate(_ update: @escaping Update<State>) {
-        pendingUpdates.append(update)
+        updateState(using: update, apply: false)
     }
-    
-    private func applyUpdates() {
-        let newState = pendingUpdates.reduce(into: state) { state, update in
-            update(&state)
+
+    private func updateState(using update: @escaping Update<State>, apply: Bool) {
+        var newState: State?
+        var pendingUpdates: [Update<State>]!
+        queue.sync {
+            self.pendingUpdates.append(update)
+            pendingUpdates = self.pendingUpdates
+            if apply {
+                self.pendingUpdates = []
+            }
+        }
+        if apply {
```

Comment:
What would you think about using an optional `pendingUpdates`?
```suggestion
        var pendingUpdates: [Update<State>]?
        queue.sync {
            self.pendingUpdates.append(update)
            if apply {
                pendingUpdates = self.pendingUpdates
                self.pendingUpdates = []
            }
        }
        if let pendingUpdates {
```

Reply:
Make sense, since we don't need make unnessesarly assignment on line 107, if we not gonna apply them